pkg/solana/txm/pendingtx.go

```diff
@@ -66,14 +75,21 @@ type pendingTx struct {
 // finishedTx is used to store minimal info specifically for finalized or errored transactions for external status checks
 type finishedTx struct {
 	retentionTs time.Time
-	state       utils.TxState
+	state       txmutils.TxState
+}
+
+type txInfo struct {
+	// id of the transaction
+	id string
+	// state of the signature
+	state txmutils.TxState
 }
 
 var _ PendingTxContext = &pendingTxContext{}
 
 type pendingTxContext struct {
-	cancelBy map[string]context.CancelFunc
-	sigToID  map[solana.Signature]string
+	cancelBy    map[string]context.CancelFunc
+	sigToTxInfo map[solana.Signature]txInfo
```

Comment:
Why not just use the existing state maps? If these diverge it could be an issue so wouldn't it be better to have a singular source of truth?